sleepyq/__init__.py

```diff
@@ -36,66 +36,208 @@ def __init__(self, data):
+        else:
+            r = self._session.get(self._api+url)
+        if r.status_code == 401: # HTTP error 401 Unauthorized
+            # Login
+            self.login()
+            # Retry Request
+            if mode == 'put':
+                r = self._session.put(self._api+url, json=data)
+            else:
+                r = self._session.get(self._api+url)
+        if r.status_code != 200: # If status code is not 200 OK
+            # Raise error
+            r.raise_for_status()
+        return r
 
     def login(self):
         if '_k' in self._session.params:
             del self._session.params['_k']
-
         data = {'login': self._login, 'password': self._password}
-        r = self._session.put('https://api.sleepiq.sleepnumber.com/rest/login', json=data)
-        r.raise_for_status()
-
+        r = self._session.put(self._api+'/login', json=data)
+        if r.status_code == 401:
+            return False
         self._session.params['_k'] = r.json()['key']
-
         return True
 
     def sleepers(self):
-        url = 'https://api.sleepiq.sleepnumber.com/rest/sleeper'
-        r = self._session.get(url)
-        r.raise_for_status()
-
+        url = '/sleeper'
+        r=self.__makeRequest(url)
         sleepers = [Sleeper(sleeper) for sleeper in r.json()['sleepers']]
         return sleepers
 
     def beds(self):
-        url = 'https://api.sleepiq.sleepnumber.com/rest/bed'
-        r = self._session.get(url)
-        r.raise_for_status()
+        url = '/bed'
+        r=self.__makeRequest(url)
         beds = [Bed(bed) for bed in r.json()['beds']]
         return beds
 
     def beds_with_sleeper_status(self):
         beds = self.beds()
         sleepers = self.sleepers()
         family_statuses = self.bed_family_status()
-
         sleepers_by_id = {sleeper.sleeper_id: sleeper for sleeper in sleepers}
         bed_family_statuses_by_bed_id = {family_status.bed_id: family_status for family_status in family_statuses}
-
         for bed in beds:
             family_status = bed_family_statuses_by_bed_id[bed.bed_id]
-
             for side in ['left', 'right']:
                 sleeper_key = 'sleeper_' + side + '_id'
                 sleeper_id = getattr(bed, sleeper_key)
+                if sleeper_id == "0":
+                    continue
                 sleeper = sleepers_by_id[sleeper_id]
-
                 status = getattr(family_status, side)
                 status.sleeper = sleeper
-
                 setattr(bed, side, status)
         return beds
 
 
     def bed_family_status(self):
-        url = 'https://api.sleepiq.sleepnumber.com/rest/bed/familyStatus'
-        r = self._session.get(url)
-        r.raise_for_status()
-
+        url = '/bed/familyStatus'
+        r=self.__makeRequest(url)
         statuses = [FamilyStatus(status) for status in r.json()['beds']]
         return statuses
+
+    def set_light(self, bedId, light, setting):
+        #
+        # light 1-4
+        ### 1=Right Night Stand
```

Comment:
Maybe we could use some constants for this? That'd let you call like:

```python
client.set_light(RIGHT_NIGHT_STAND)
```

This applies to a few other places, like `get_light`, and in different contexts (`preset`)

Reply:
I created some constants, but not sure if I put them in the correct location.